share/spice/icon/icon.js

```diff
@@ -0,0 +1,90 @@
+(function (env) {
+    "use strict";
+    env.ddg_spice_icon = function(api_result){
+
+        // Validate the response (customize for your Spice)
+        if (!api_result || api_result.error || api_result.icons.length == 0) {
+            return Spice.failed('icon');
+        }
+
+        var script = $('[src*="/js/spice/icon/"]')[0],
+            source = $(script).attr("src"),
+            query = source.match(/icon\/([^\/]+)/)[1],
+            decodedQuery = decodeURIComponent(query);
+
+        // Render the response
+        Spice.add({
+            id: "icon",
+
+            // Customize these properties
+            name: "Icon",
+            data: api_result.icons,
+            meta: {
+                itemType: 'Icons',
+                searchTerm: decodedQuery,
+                sourceName: "iconfinder.com",
```

Comment:
Change to "Iconfinder" instead of "iconfinder.com". We usually go with the title of the website like in https://duckduckgo.com/?q=ray+bradbury+books&ia=products we say "Amazon" instead of "amazon.com".